File: backend/app/models/game_definition.py
"""GameDefinition model.

Defines a cognitive *exercise* (not a medical test). Names/slugs must be neutral
exercise labels (e.g. memory_match, attention_focus, reaction_time,
sequence_order). No diagnostic naming or medical interpretation.
"""

from typing import Optional

from sqlalchemy import Boolean, Integer, String, Text
from sqlalchemy.orm import Mapped, mapped_column

from app.db.base import Base
from app.db.mixins import SoftDeleteMixin, TimestampMixin, UUIDPrimaryKeyMixin


class GameDefinition(UUIDPrimaryKeyMixin, TimestampMixin, SoftDeleteMixin, Base):
    __tablename__ = "game_definitions"

    name: Mapped[str] = mapped_column(String(255), nullable=False)
    slug: Mapped[str] = mapped_column(String(64), unique=True, nullable=False)
    description: Mapped[Optional[str]] = mapped_column(Text, nullable=True)
    game_type: Mapped[str] = mapped_column(String(64), nullable=False)
    difficulty: Mapped[str] = mapped_column(String(32), default="easy", nullable=False)
    estimated_duration_minutes: Mapped[Optional[int]] = mapped_column(
        Integer, nullable=True
    )
    active: Mapped[bool] = mapped_column(Boolean, default=True, nullable=False)
    instructions: Mapped[Optional[str]] = mapped_column(Text, nullable=True)

    def __repr__(self) -> str:  # pragma: no cover - debug helper
        return f"<GameDefinition id={self.id!r} slug={self.slug!r}>"

File: mobile/lib/features/games/data/game_definition.dart
/// A cognitive-exercise game definition (from GET /api/v1/games).
///
/// Exercise metadata only — no diagnostic or medical fields.
class GameDefinition {
  const GameDefinition({
    required this.id,
    required this.name,
    required this.slug,
    required this.gameType,
    required this.difficulty,
    this.description,
    this.instructions,
    this.estimatedDurationMinutes,
    this.active = true,
  });

  final String id;
  final String name;
  final String slug;
  final String gameType;
  final String difficulty;
  final String? description;
  final String? instructions;
  final int? estimatedDurationMinutes;
  final bool active;

  factory GameDefinition.fromJson(Map<String, dynamic> json) {
    return GameDefinition(
      id: (json['id'] ?? '').toString(),
      name: (json['name'] as String?) ?? '',
      slug: (json['slug'] as String?) ?? '',
      gameType: (json['game_type'] as String?) ?? '',
      difficulty: (json['difficulty'] as String?) ?? '',
      description: json['description'] as String?,
      instructions: json['instructions'] as String?,
      estimatedDurationMinutes:
          (json['estimated_duration_minutes'] as num?)?.toInt(),
      active: (json['active'] as bool?) ?? true,
    );
  }
}
